File: test_data/code/java/1184-distance-between-bus-stops.java
class Solution {
    public int distanceBetweenBusStops(int[] distance, int start, int destination) {
        int total = 0;
        int min = Integer.MAX_VALUE;
        for (int i = 1; i < distance.length; i++) {
            total += distance[i];
            min = Math.min(min, distance[i]);
        }
        total += distance[0];
        total = Math.min(total, total - (distance[destination] + total - distance[start]));
        return Math.min(total, total - min);
    }
}

File: test_data/code/python/1184-distance-between-bus-stops.py
class Solution:
    def distanceBetweenBusStops(self, distance: List[int], start: int, destination: int) -> int:
        # Calculate total distance to find nearest multiple of destination - start
        total_distance = sum(distance)
        # First try going in the forward direction
        forward_distance = sum(distance[start:destination % len(distance) + 1])
        # Second try going in the backward direction
        backward_distance = total_distance - forward_distance
        # Return the smaller of the two distances
        return min(forward_distance, backward_distance)